### PR TITLE
[WIP] JS2 support

### PR DESCRIPTION
**Pull request by almahmoud**
Friends with https://github.com/galaxyproject/cloudlaunch/pull/260 https://github.com/CloudVE/cloudbridge/pull/297  https://github.com/CloudVE/djcloudbridge/pull/10

### CODE REVIEW

File: roles/rke/templates/ebs_storage_class.j2 [hunk 1]
```diff
@@ -39,7 +39,7 @@ provisioner: kubernetes.io/cinder
 provisioner: cinder.csi.openstack.org
 {% endif %}
 parameters:
-  availability: melbourne-qh2
+  availability: nova
```
Comment (nuwang) [inline roles/rke/templates/ebs_storage_class.j2:42]: Can we parameterize this?

File: roles/rke/tasks/registration.yaml [hunk 2]
```diff
@@ -26,9 +26,9 @@
 - name: Install OpenStack Cloud Controller Manager
   shell: kubectl apply -f {{ item }}
   with_items:
-    - https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/cluster/addons/rbac/cloud-controller-manager-roles.yaml
-    - https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/cluster/addons/rbac/cloud-controller-manager-role-bindings.yaml
-    - https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/openstack-cloud-controller-manager-ds.yaml
+    - https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/cloud-controller-manager-roles.yaml
+    - https://raw.githubusercontent.com/kubernetes/cloud-provider-openstack/master/manifests/controller-manager/cloud-controller-manager-role-bindings.yaml
+    - https://gist.githubusercontent.com/almahmoud/b02942eb2b18ea14123e163e069a79e2/raw/b59139b17d697c9075ce3870ad7e88c567851d8a/openstack-ds.yaml
```
Comment (nuwang) [inline roles/rke/tasks/registration.yaml:31]: Why is the custom gist needed? Can we check this in somewhere, maybe into the cloudman-boot repo itself?

File: roles/rke/tasks/common.yaml [hunk 3]
```diff
@@ -101,5 +101,5 @@
   wait_for:
     path: /etc/rancher/node/password
     state: present
-    timeout: 180
+    timeout: 1800
```
Comment (nuwang) [inline roles/rke/tasks/common.yaml:104]: Half an hour for the password seems a fairly long time. Does it take that long?

File: roles/rke/defaults/main.yml [hunk 4]
```diff
@@ -4,7 +4,7 @@
 kube_cloud_provider: "{{ lookup('env', 'KUBE_CLOUD_PROVIDER') }}"
 kube_cloud_conf: "{{ lookup('env', 'KUBE_CLOUD_CONF') | b64decode }}"
 # Whether to use the in-tree cloud provider or external cloud-provider
-kube_in_tree_provider: true
+kube_in_tree_provider: false
```
Comment (nuwang) [inline roles/rke/defaults/main.yml:7]: Won't this break the other clouds?  Did you check the csi drivers as well?

File: roles/rke/tasks/common.yaml (hunk 3 above)
Comment (almahmoud) [reply roles/rke/tasks/common.yaml:104]: It did on JS1 on TACC, this is an old change

File: roles/rke/defaults/main.yml (hunk 4 above)
Comment (almahmoud) [reply roles/rke/defaults/main.yml:7]: CSI drivers work for openstack JS2. Haven't looked at other clouds, but I also imagine they're a lot less needed given they have k8s services, so at least personally I think I'm only using this stack for JS

File: roles/rke/tasks/registration.yaml (hunk 2 above)
Comment (almahmoud) [reply roles/rke/tasks/registration.yaml:31]: The control-plane vs master change for the daemonset node selector was the main issue. The custom gist was to make it work with 1.19 as before. With 1.24, the newer one from their repo works, but the patch under needs to change, and helm versions (notably for nginx) need to be upgraded. This was the easiest way to get it working in the meantime

**Comment by nuwang** [comment]
I didn't have access to the bioconductor source branch, so I merged this separately.

**Pull request closed without merging**